io/src/main/scala/sbt/internal/io/Observers.scala

```diff
@@ -0,0 +1,161 @@
+/*
+ * sbt IO
+ *
+ * Copyright 2011 - 2019, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ */
+
+package sbt.internal.io
+
+import java.io.IOException
+import java.nio.file.{ Path => NioPath }
+import java.util
+import java.util.concurrent.atomic.AtomicInteger
+
+import sbt.io.Glob
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+import scala.util.control.NonFatal
+
+private[sbt] trait Observer[-T] extends AutoCloseable {
+
+  /**
+   * Run a callback
+   * @param t the generic type of events
+   */
+  def onNext(t: T): Unit
+
+  /**
+   * This is for managed observers that may need to be removed once the caller is done with them.
+   * See [[sbt.internal.io.Observers]].
+   */
+  override def close(): Unit = {}
+}
+
+/**
+ * An object that monitors a file system. The interface is very similar to that provided by other
+ * libraries/frameworks, such as [[http://reactivex.io/intro.html rxJava]]. When it detects changes
+ * in the file system, it will invoke a set of user specified callbacks. The Observable also
+ * allows the user to add and removes paths to monitor.
+ *
+ * @tparam T the generic type of observable values
+ */
+private[sbt] trait Observable[+T] extends AutoCloseable {
```

Comment:
Is RxJava style APIs getting a promotion here by moving it to top level?
It might be helpful to note your current thinking about its shelf life.

Reply:
It still in the internal package and private sbt so I don’t really think of this as a promotion. I’m not sure if it will survive in the long run depending on how the replacement file system pans out.